src/Admin/Extensions/Extensions.php

```diff
@@ -205,139 +204,114 @@ private function get_installed_plugins() {
+			'plugin_url',
+			'support_url',
+			'documentation_url',
+		];
+		foreach ( $required_fields as $property ) {
+			if ( empty( $extension[ $property ] ) ) {
+				return new \WP_Error(
+					$error_code,
+					sprintf( $error_message, $extension['name'], $property )
+				);
 			}
-			$this->schema = json_decode( $schema_file );
 		}
 
-		// Validate the extension data based on the loaded schema
-		$validator = new \JsonSchema\Validator();
-		$validator->validate( $extension, $this->schema );
-
-		if ( ! $validator->isValid() ) {
-			return $validator->getErrors()[0]['message'] ?? 'Unknown error';
+		// Ensure Author has the required name field.
+		if ( empty( $extension['author']['name'] ) ) {
+			return new \WP_Error(
+				$error_code,
+				sprintf( $error_message, $extension['name'], 'author.name' )
+			);
 		}
 
 		return true;
 	}
 
 	/**
-	 * Get the list of WPGraphQL extensions.
+	 * Populate the extensions list with installation data.
 	 *
-	 * @return array<int, array<string, mixed>> List of extensions.
+	 * @param Extension[] $extensions The extensions to populate.
+	 *
+	 * @return PopulatedExtension[] The populated extensions.
 	 */
-	public function get_extensions(): array {
+	private function populate_installation_data( $extensions ) {
+		$installed_plugins = $this->get_installed_plugins();
 
-		$valid_extensions = [];
+		$populated_extensions = [];
 
-		if ( null === $this->extensions ) {
-
-			// Allow filtering the extensions before they are loaded
-			$pre_filtered_extensions = apply_filters( 'graphql_pre_get_extensions', null );
-			if ( null !== $pre_filtered_extensions ) {
-				$extensions = $pre_filtered_extensions;
-			} else {
-				$extensions = $this->load_extensions_from_file() ?? [];
-			}
-
-			$installed_plugins = $this->get_installed_plugins();
-
-			// Filter the list of extensions, allowing other plugins to add their own extensions
-			$extensions = apply_filters( 'graphql_get_extensions', $extensions );
-
-			foreach ( $extensions as $extension ) {
-				if ( true === $this->is_valid_extension( $extension ) ) {
-					$valid_extensions[] = $extension;
-				}
-			}
-
-			// Validate and remove invalid extensions
-			$this->extensions = ! empty( $valid_extensions ) ? $valid_extensions : [];
-		}
-
-		if ( empty( $this->extensions ) ) {
-			return [];
-		}
-
-		foreach ( $this->extensions as $extension ) {
+		foreach ( $extensions as $extension ) {
 			$slug = basename( rtrim( $extension['plugin_url'], '/' ) );
+			$extension['installed'] = false;
+			$extension['active']    = false;
+
+			// If the plugin is installed, populate the installation data.
 			if ( isset( $installed_plugins[ $slug ] ) ) {
 				$extension['installed'] = true;
 				$extension['active']    = $installed_plugins[ $slug ]['is_active'];
 				$extension['author']    = $installed_plugins[ $slug ]['author'];
-			} else {
-				$extension['installed'] = false;
-				$extension['active']    = false;
 			}
 
+			// @todo Where does this come from?
```

Comment:
I'm not seeing `extension['setting_path']` populated anywhere or noted in any of the docs.

Reply:
@justlevine ah yes, this was one that we had at one point, but removed it and didn't clean up 🤦🏻‍♂️

Reply:
Another reasons to keep the validation coupled with the code in PHP and not in an external schema ;-)

Reply:
ha! For sure. 

Here was my thoughts on why I was using .json files: 

https://github.com/wp-graphql/wp-graphql/pull/3246#issuecomment-2515620055

Reply:
Ok, ya, the more I think about this, the more I like the simplicity. I was way over-complicating, because of my "need" to use the same list of extensions as localized data for the react extensions page in the plugin but also within the WPGraphQL.com site. 

In my mind we needed something that could be read by both JS and PHP and .json seemed to be the right approach. 

This simplifies things, and now I will just need to get the extensions data for display on wpgraphql.com, but since it already sources Blog Posts, etc from a WordPress server, it can easily source this list from a WordPress server as well, it doesn't need to source it from the Github repo, which was my initial plan.